### PR TITLE
[BugFix] fix duplicate array_agg/group_concat in CallOperator

### LINKED ISSUE #31757: Multiple ARRAY_AGG+order by results in wrong ordering direction
### Steps to reproduce the behavior (Required)
```
admin set frontend config ('enable_statistic_collect_on_first_load'='false');

CREATE DATABASE IDS_AGGREGATION_TRUE;

CREATE TABLE  `IDS_AGGREGATION_TRUE` . `ACTIVITY`  ( `case`  BIGINT,  `timestamp`  DATETIME,  `ID_ACTIVITY`  BIGINT,  `activity`  BIGINT,  `val`  BIGINT,  `strval`  STRING,  `dateval`  DATETIME,  `activityname`  STRING,  `FK_8_ACTIVITY_CASE`  BIGINT) PRIMARY KEY( `case` , `timestamp` , `ID_ACTIVITY` ) DISTRIBUTED BY HASH ( `case` ) PROPERTIES("replication_num" = "1");
CREATE TABLE  `IDS_AGGREGATION_TRUE` . `CASE`  ( `ID_CASE`  BIGINT,  `case`  BIGINT,  `val`  BIGINT,  `FK_8_ACTIVITY_CASE`  BIGINT,  `PU_9_ARRAY_AGG_8_activity_case`  ARRAY<BIGINT>,  `PU_9_ARRAY_AGG_8_activity_timestamp`  ARRAY<DATETIME>,  `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY`  ARRAY<BIGINT>,  `PU_9_ARRAY_AGG_8_activity_activity`  ARRAY<BIGINT>,  `PU_9_ARRAY_AGG_8_activity_val`  ARRAY<BIGINT>,  `PU_9_ARRAY_AGG_8_activity_strval`  ARRAY<STRING>,  `PU_9_ARRAY_AGG_8_activity_dateval`  ARRAY<DATETIME>,  `PU_9_ARRAY_AGG_8_activity_activityname`  ARRAY<STRING>,  `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE`  ARRAY<BIGINT>) PRIMARY KEY( `ID_CASE` ) DISTRIBUTED BY HASH ( `ID_CASE` ) PROPERTIES("replication_num" = "1");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(1,1,"1992-09-20 11:30:01.000","1",2,NULL,32,1,"1992-09-20 11:30:01.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(2,2,"1992-09-20 11:30:02.000","2",4,NULL,32,1,"1992-09-20 11:30:02.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(2,3,"1992-09-20 11:30:02.000","2",6,NULL,32,1,"1992-09-20 11:30:03.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(2,4,"1992-09-20 11:30:02.000","2",8,NULL,32,1,"1992-09-20 11:30:04.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(3,5,"1992-09-20 11:30:03.000","3",10,NULL,33,2,"1992-09-20 11:30:05.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(3,6,"1992-09-20 11:30:03.000","3",12,NULL,33,2,"1992-09-20 11:30:06.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(4,7,"1992-09-20 11:30:04.000","4",14,NULL,33,2,"1992-09-20 11:30:07.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(4,8,"1992-09-20 11:30:04.000","4",16,NULL,33,2,"1992-09-20 11:30:08.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(5,9,"1992-09-20 11:30:05.000","5",18,NULL,34,3,"1992-09-20 11:30:09.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(6,10,"1992-09-20 11:30:06.000","6",20,NULL,34,3,"1992-09-20 11:30:10.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(200,11,"1992-09-20 11:32:00.000","200",22,NULL,34,3,"1992-09-20 11:30:11.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(201,12,"1992-09-20 11:32:01.000","201",24,NULL,34,3,"1992-09-20 11:30:12.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(NULL,13,NULL,NULL,26,NULL,35,4,"1992-09-20 11:30:13.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(6,14,"1992-09-20 11:30:06.000","6",28,NULL,35,4,"1992-09-20 11:30:14.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(201,16,"1992-09-20 11:32:01.000","201",30,NULL,35,4,"1992-09-20 11:30:15.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(6,17,"1992-09-20 11:30:06.000","6",32,NULL,36,5,"1992-09-20 11:30:16.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(201,19,"1992-09-20 11:32:01.000","201",34,NULL,36,5,"1992-09-20 11:30:17.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(NULL,20,NULL,NULL,36,NULL,36,5,"1992-09-20 11:30:18.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(NULL,21,NULL,NULL,38,NULL,37,6,"1992-09-20 11:30:19.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(NULL,22,NULL,NULL,40,NULL,37,6,"1992-09-20 11:30:20.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(NULL,23,NULL,NULL,42,NULL,37,6,"1992-09-20 11:30:21.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(NULL,24,NULL,NULL,44,NULL,37,6,"1992-09-20 11:30:22.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(10,25,"1992-09-20 11:30:10.000","10",46,NULL,38,7,"1992-09-20 11:30:23.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(20,26,"1992-09-20 11:30:20.000","20",48,NULL,38,7,"1992-09-20 11:30:24.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(30,27,"1992-09-20 11:30:30.000","30",50,NULL,38,7,"1992-09-20 11:30:25.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(40,28,"1992-09-20 11:30:40.000","40",52,NULL,38,7,"1992-09-20 11:30:26.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(10,29,"1992-09-20 11:30:10.000","10",54,NULL,39,8,"1992-09-20 11:30:27.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(20,30,"1992-09-20 11:30:20.000","20",56,NULL,39,8,"1992-09-20 11:30:28.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(30,31,"1992-09-20 11:30:30.000","30",58,NULL,39,8,"1992-09-20 11:30:29.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(40,32,"1992-09-20 11:30:40.000","40",60,NULL,39,8,"1992-09-20 11:30:30.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `ACTIVITY` ( `val` , `activity` , `dateval` , `strval` , `ID_ACTIVITY` , `activityname` , `FK_8_ACTIVITY_CASE` , `case` , `timestamp` ) VALUES(50,33,"1992-09-20 11:30:50.000","50",62,NULL,39,8,"1992-09-20 11:30:31.000");
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(100,["1992-09-20 11:30:01.000", "1992-09-20 11:30:02.000", "1992-09-20 11:30:03.000", "1992-09-20 11:30:04.000"],[1, 2, 2, 2],[1, 2, 3, 4],[1, 1, 1, 1],["1992-09-20 11:30:01.000", "1992-09-20 11:30:02.000", "1992-09-20 11:30:02.000", "1992-09-20 11:30:02.000"],[32, 32, 32, 32],[NULL, NULL, NULL, NULL],[2, 4, 6, 8],["1", "2", "2", "2"],32,1,2);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(200,["1992-09-20 11:30:05.000", "1992-09-20 11:30:06.000", "1992-09-20 11:30:07.000", "1992-09-20 11:30:08.000"],[3, 3, 4, 4],[5, 6, 7, 8],[2, 2, 2, 2],["1992-09-20 11:30:03.000", "1992-09-20 11:30:03.000", "1992-09-20 11:30:04.000", "1992-09-20 11:30:04.000"],[33, 33, 33, 33],[NULL, NULL, NULL, NULL],[10, 12, 14, 16],["3", "3", "4", "4"],33,2,4);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(300,["1992-09-20 11:30:09.000", "1992-09-20 11:30:10.000", "1992-09-20 11:30:11.000", "1992-09-20 11:30:12.000"],[5, 6, 200, 201],[9, 10, 11, 12],[3, 3, 3, 3],["1992-09-20 11:30:05.000", "1992-09-20 11:30:06.000", "1992-09-20 11:32:00.000", "1992-09-20 11:32:01.000"],[34, 34, 34, 34],[NULL, NULL, NULL, NULL],[18, 20, 22, 24],["5", "6", "200", "201"],34,3,6);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(400,["1992-09-20 11:30:13.000", "1992-09-20 11:30:14.000", "1992-09-20 11:30:15.000"],[NULL, 6, 201],[13, 14, 16],[4, 4, 4],[NULL, "1992-09-20 11:30:06.000", "1992-09-20 11:32:01.000"],[35, 35, 35],[NULL, NULL, NULL],[26, 28, 30],[NULL, "6", "201"],35,4,8);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(500,["1992-09-20 11:30:16.000", "1992-09-20 11:30:17.000", "1992-09-20 11:30:18.000"],[6, 201, NULL],[17, 19, 20],[5, 5, 5],["1992-09-20 11:30:06.000", "1992-09-20 11:32:01.000", NULL],[36, 36, 36],[NULL, NULL, NULL],[32, 34, 36],["6", "201", NULL],36,5,10);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(600,["1992-09-20 11:30:19.000", "1992-09-20 11:30:20.000", "1992-09-20 11:30:21.000", "1992-09-20 11:30:22.000"],[NULL, NULL, NULL, NULL],[21, 22, 23, 24],[6, 6, 6, 6],[NULL, NULL, NULL, NULL],[37, 37, 37, 37],[NULL, NULL, NULL, NULL],[38, 40, 42, 44],[NULL, NULL, NULL, NULL],37,6,12);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(700,["1992-09-20 11:30:23.000", "1992-09-20 11:30:24.000", "1992-09-20 11:30:25.000", "1992-09-20 11:30:26.000"],[10, 20, 30, 40],[25, 26, 27, 28],[7, 7, 7, 7],["1992-09-20 11:30:10.000", "1992-09-20 11:30:20.000", "1992-09-20 11:30:30.000", "1992-09-20 11:30:40.000"],[38, 38, 38, 38],[NULL, NULL, NULL, NULL],[46, 48, 50, 52],["10", "20", "30", "40"],38,7,14);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `PU_9_ARRAY_AGG_8_activity_timestamp` , `PU_9_ARRAY_AGG_8_activity_val` , `PU_9_ARRAY_AGG_8_activity_activity` , `PU_9_ARRAY_AGG_8_activity_case` , `PU_9_ARRAY_AGG_8_activity_dateval` , `PU_9_ARRAY_AGG_8_activity_FK_8_ACTIVITY_CASE` , `PU_9_ARRAY_AGG_8_activity_activityname` , `PU_9_ARRAY_AGG_8_activity_ID_ACTIVITY` , `PU_9_ARRAY_AGG_8_activity_strval` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(800,["1992-09-20 11:30:27.000", "1992-09-20 11:30:28.000", "1992-09-20 11:30:29.000", "1992-09-20 11:30:30.000", "1992-09-20 11:30:31.000"],[10, 20, 30, 40, 50],[29, 30, 31, 32, 33],[8, 8, 8, 8, 8],["1992-09-20 11:30:10.000", "1992-09-20 11:30:20.000", "1992-09-20 11:30:30.000", "1992-09-20 11:30:40.000", "1992-09-20 11:30:50.000"],[39, 39, 39, 39, 39],[NULL, NULL, NULL, NULL, NULL],[54, 56, 58, 60, 62],["10", "20", "30", "40", "50"],39,8,16);
INSERT INTO  `IDS_AGGREGATION_TRUE` . `CASE` ( `val` , `FK_8_ACTIVITY_CASE` , `case` , `ID_CASE` ) VALUES(900,40,9,18);
```

```
WITH `ParserContext_Table_ACTIVITY_000` AS (
  SELECT
    `TableScan_Alias_ACTIVITY_001`.`ACTIVITY` AS `JoinColumnResolver_Column_ACTIVITY_000`,
    `ParserContext_Alias_CASE_000`.`CASE` AS `JoinColumnResolver_Column_CASE_001`,
    `TableScan_Alias_ACTIVITY_001`.`STRVAL` AS `JoinColumnResolver_Column_STRVAL_000`
  FROM
    `IDS_AGGREGATION_TRUE`.`ACTIVITY` AS `TableScan_Alias_ACTIVITY_001`
    LEFT JOIN `IDS_AGGREGATION_TRUE`.`CASE` AS `ParserContext_Alias_CASE_000` ON (
      `TableScan_Alias_ACTIVITY_001`.`FK_8_ACTIVITY_CASE` = `ParserContext_Alias_CASE_000`.`FK_8_ACTIVITY_CASE`
    )
),
`ParserContext_Aggregation_000` AS (
  SELECT
    ARRAY_JOIN(
      ARRAY_AGG(
        `TableScan_Alias_Flattener_000`.`JoinColumnResolver_Column_STRVAL_000`
        ORDER BY
          `TableScan_Alias_Flattener_000`.`JoinColumnResolver_Column_ACTIVITY_000` DESC
      ),
      ":"
    ) AS `AggregateComputationsHandler_AGG_000`,
    ARRAY_JOIN(
      ARRAY_AGG(
        `TableScan_Alias_Flattener_000`.`JoinColumnResolver_Column_STRVAL_000`
        ORDER BY
          `TableScan_Alias_Flattener_000`.`JoinColumnResolver_Column_ACTIVITY_000` ASC
      ),
      ":"
    ) AS `AggregateComputationsHandler_AGG_001`,
    `TableScan_Alias_Flattener_000`.`JoinColumnResolver_Column_CASE_001` AS `JoinColumnResolver_Column_CASE_001`
  FROM
    `ParserContext_Table_ACTIVITY_000` AS `TableScan_Alias_Flattener_000`
  GROUP BY
    `TableScan_Alias_Flattener_000`.`JoinColumnResolver_Column_CASE_001`
)
SELECT * from ParserContext_Aggregation_000;
```

### Expected behavior (Required)
```
40:30:20:10	10:20:30:40	7
2:2:2:1	1:2:2:2	1
4:4:3:3	3:3:4:4	2
50:40:30:20:10	10:20:30:40:50	8
201:6	6:201	4
		6
201:6	6:201	5
201:200:6:5	5:6:200:201	3
```

### Real behavior (Required)
```
40:30:20:10	40:30:20:10	7
2:2:2:1	2:2:2:1	1
4:4:3:3	4:4:3:3	2
50:40:30:20:10	50:40:30:20:10	8
201:6	201:6	4
		6
201:6	201:6	5
201:200:6:5	201:200:6:5	3
```

### StarRocks version (Required)
starrocks/allin1-ubuntu:3.1.0-rc01

### PR DESCRIPTION
**Pull request by fzhedu**
Fixes https://github.com/StarRocks/starrocks/issues/31757
equal `array_agg/group_concat `with consideration of nullsFist and isAsc when compare `CallOperator`

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/catalog/AggregateFunction.java [hunk 1]
```diff
@@ -330,6 +332,21 @@ public String toSql(boolean ifNotExists) {
...
+            return false;
+        }
+        AggregateFunction agg = (AggregateFunction) obj;
+        if (!Objects.equals(intermediateType, agg.intermediateType) || ignoresDistinct != agg.ignoresDistinct ||
+                isAnalyticFn != agg.isAggregateFn || returnsNonNullOnEmpty != agg.returnsNonNullOnEmpty ||
+                !Objects.equals(symbolName, agg.symbolName) || isDistinct != agg.isDistinct ||
+                !Objects.equals(nullsFirst, agg.getNullsFirst()) || !Objects.equals(isAscOrder, agg.getIsAscOrder())) {
+            return false;
+        }
+        return super.equals(obj);
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/catalog/AggregateFunction.java:347]: It seems not easy to read.  How about firstly satisfied `super.equals(obj)` then do the private field comparing process.

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by packy92 — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 9 / 9 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/AggregateFunction.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/CallOperator.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:heart_eyes: **pass** : 11 / 11 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/AggregateFunction.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/scalar/CallOperator.java|4|4|100.00%|[]|

**Review by packy92 — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**